homeassistant/components/fan/__init__.py

```diff
@@ -237,17 +239,60 @@ def set_preset_mode(self, preset_mode: str) -> None:
         """Set new preset mode."""
         raise NotImplementedError()
 
+    async def _async_set_preset_mode(self, preset_mode: str) -> None:
+        """Validate and set new preset mode."""
+        self.valid_preset_mode_or_raise(preset_mode)
+        await self.async_set_preset_mode(preset_mode)
+
     async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set new preset mode."""
         await self.hass.async_add_executor_job(self.set_preset_mode, preset_mode)
 
+    @callback
     def _valid_preset_mode_or_raise(self, preset_mode: str) -> None:
+        """Raise NotValidPresetModeError on invalid preset_mode."""
+        integration_platform: str = "unknown"
+        if (entry := self.registry_entry) is not None:
+            integration_platform = entry.platform
+
+        _LOGGER.warning(
+            "The integration %s implements "
+            "entity._valid_preset_mode_or_raise, which is is deprecated. Support "
+            "will be removed with HA Core 2024.12. "
+            "The fan entity component already validates the preset_mode "
+            "when fan.async_turn_on or fan_entity._async_set_preset_mode is called. "
+            "If still needed, the function fan_entity.valid_preset_mode_or_raise "
+            "can be used instead",
+            integration_platform,
+        )
+        async_create_issue(
+            self.hass,
+            DOMAIN,
+            f"deprecated_preset_mode_validator_{integration_platform}",
+            breaks_in_ha_version="2024.12.0",
+            is_fixable=False,
+            translation_key="deprecated_preset_mode_validator",
+            translation_placeholders={"integration": integration_platform},
+            severity=IssueSeverity.WARNING,
+        )
+        self.valid_preset_mode_or_raise(preset_mode)
+
+    @final
+    @callback
+    def valid_preset_mode_or_raise(self, preset_mode: str) -> None:
```

Comment:
I'd now keep this protected and rename it to make it clear it's temporary. When we remove `_valid_preset_mode_or_raise` we can inline this into `_async_turn_on`.

Reply:
Do we still want to register an issue?

Reply:
> I'd now keep this protected and rename it to make it clear it's temporary. When we remove `_valid_preset_mode_or_raise` we can inline this into `_async_turn_on`.

The validator also used before calling `entity.async_set_preset_mode()`

Reply:
If we rename it, we might consider to re-add it as a catcher for custom integrations that might have implemented it and might break.